### ISSUE TITLE
🐛 BUG: Astro crashes on nested scripts

### ISSUE THREAD

**Issue by thepassle** [snowpackjs/astro#742]
### What package manager are you using?

npm

### What operating system are you using?

mac

### Describe the Bug

In my docs site, I like to use prism (note: not the built in astro Prism component) to display how a user may load my library, e.g.:

![image](https://user-images.githubusercontent.com/17054057/126081159-f9e96998-c85a-4cbe-9eb0-df974aa48f83.png)

The markup to display this looks like:
```html
  <pre><code class="language-markup"><script type="prism-html-markup"><script src="https://unpkg.com/@generic-components/components@1.0.0/switch.js" type="module"></script></script></code></pre>
```

Which, admittedly, is a little odd, but just how prism works I suppose. 

Astro seems to be tripping up on the nested `<script>` tags, and throws an error:

```
[23:23:09] [snowpack] Build Result Error: There was a problem with a file build result.
[executing astro] CompileError [ParseError]: </script> attempted to close an element that was not open
    at error (/Users/au87xu/generic-components/node_modules/@astrojs/parser/src/utils/error.ts:40:15)
    at Parser.error (/Users/au87xu/generic-components/node_modules/@astrojs/parser/src/parse/index.ts:109:5)
    at tag (/Users/au87xu/generic-components/node_modules/@astrojs/parser/src/parse/state/tag.ts:167:16)
    at new Parser (/Users/au87xu/generic-components/node_modules/@astrojs/parser/src/parse/index.ts:55:15)
    at parse (/Users/au87xu/generic-components/node_modules/@astrojs/parser/src/parse/index.ts:227:18)
    at convertAstroToJsx (file:///Users/au87xu/generic-components/node_modules/astro/src/compiler/index.ts:31:15)
    at transformFromSource (file:///Users/au87xu/generic-components/node_modules/astro/src/compiler/index.ts:91:20)
    at compileComponent (file:///Users/au87xu/generic-components/node_modules/astro/src/compiler/index.ts:110:24)
    at Object.load (/Users/au87xu/generic-components/node_modules/astro/snowpack-plugin.cjs:68:28)
    at runPipelineLoadStep (/Users/au87xu/generic-components/node_modules/snowpack/lib/index.js:123895:32)
    at Object.buildFile (/Users/au87xu/generic-components/node_modules/snowpack/lib/index.js:124095:24)
    at /Users/au87xu/generic-components/node_modules/snowpack/lib/index.js:157676:37
    at FileBuilder.build (/Users/au87xu/generic-components/node_modules/snowpack/lib/index.js:157688:32)
    at Object.loadUrl (/Users/au87xu/generic-components/node_modules/snowpack/lib/index.js:177102:17)
    at load (/Users/au87xu/generic-components/node_modules/snowpack/lib/index.js:176699:28)
    at load (file:///Users/au87xu/generic-components/node_modules/astro/src/runtime.ts:106:17)
    at Server.<anonymous> (file:///Users/au87xu/generic-components/node_modules/astro/src/dev.ts:33:20) {
  start: { line: 19, column: 172, character: 828 },
  end: { line: 19, column: 172, character: 828 },
  filename: '/Users/au87xu/generic-components/src/pages/switch.astro',
  frame: '17:   <h2>Usage:</h2>\n' +
    '18:   <p>The switch component requires a label, which you can provide as a textnode or as an attribute. See the example down below.\n' +
    '19:   <pre><code class="language-markup"><script type="prism-html-markup"><script src="https://unpkg.com/@generic-components/components@1.0.0/switch.js" type="module"></script></script></code></pre>\n' +
    '                                                                                                                                                                                ^\n' +
    '20: \n' +
    '21:   <Demo title="Default" demo={() => (<generic-switch>Default</generic-switch>)} />',
  __snowpackBuildDetails: { name: 'snowpack-astro', step: 'load' }
}
```

### Steps to Reproduce

1. `npm init astro` using template <NAME>
2. ...
3. ...
4. ...
5. Error! Describe what went wrong (and what was expected instead)...

**Comment by FredKSchott** [comment]
Unfortunately my fix didn't work, and caused our parser to execute anything inside of the script tag as text (meaning `{}` started getting parsed). A smarter parser could handle this and I'd love a PR to take a look at this, but right now I don't think this is something we can prioritize as a team.

In the meantime, the workaround for anything like this should be:

```
<pre><code class="language-markup">{`<script...`}</code></pre>
```

We might want to consider never parsing `<code>` elements either, but for right now we assume users want to be able to use `{}` JS expressions inside of them.

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.
